docs/streams/upgrade-guide.html

```diff
@@ -173,6 +171,21 @@ <h3><a id="streams_api_changes_400" href="#streams_api_changes_400">Streams API
       See <a href="https://cwiki.apache.org/confluence/x/TZCMEw">KIP-1112</a> for more details.
     </p>
 
+    <h3><a id="streams_api_changes_400" href="#streams_api_changes_400">Streams API changes in 4.0.0</a></h3>
+
+    <p>
+        Upgraded RocksDB dependency to version 9.7.3 (from 7.9.2). This upgrade incorporates various improvements and optimizations within RocksDB. However, it also introduces some API changes.
+        The <code>org.rocksdb.AccessHint</code> class, along with its associated methods, has been removed.
+
+        <code>org.rocksdb.Options.setMemtableMaxRangeDeletions(int p1): ColumnFamilyOptionsInterface</code> has been added, providing control over the maximum number of range deletions in the memtable.
+        <code>org.rocksdb.Options.memtableMaxRangeDeletions(): int</code> has been added, allowing retrieval of the configured maximum number of range deletions in the memtable.
```

Comment:
I would only add statements about stuff that was removed because that breaks the users' applications.